rosidl_typesupport_connext_c/src/wstring_conversion.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rosidl_typesupport_connext_c/wstring_conversion.hpp>
+
+#include "ndds/ndds_c.h"
+#include "rosidl_generator_c/u16string_functions.h"
+
+namespace rosidl_typesupport_connext_c
+{
+
+DDS_Wchar * create_wstring_from_u16string(const rosidl_generator_c__U16String & u16str)
+{
+  DDS_Wchar * wstr = DDS_Wstring_alloc(u16str.size);
+  for(size_t i = 0; i < u16str.size; ++i) {
+    wstr[i] = static_cast<DDS_Wchar>(u16str.data[i]);
+  }
+  wstr[u16str.size] = static_cast<DDS_Wchar>(u'\0');
+  return wstr;
+}
+
+void wstring_to_u16string(const DDS_Wchar * wstr, rosidl_generator_c__U16String & u16str)
+{
+  size_t size = static_cast<size_t>(DDS_Wstring_length(wstr));
+  rosidl_generator_c__U16String__resize(&u16str, size);
```

Comment:
Guard against this call failing.

Reply:
Done in 761f962d4c8db310ee9f0a422d4c12c989606b7a as well as https://github.com/ros2/rosidl_typesupport_fastrtps/pull/22/commits/dc5ae8d421d52f43a82b8c29e0f35e6eada243c0.